x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -447,15 +453,15 @@ private static boolean isChildActionAuthorizedByParent(RequestInfo requestInfo,
             return false;
         }
 
+        assert Arrays.stream(indices).noneMatch(Regex::isSimpleMatchPattern)
+            || indicesRequest.indicesOptions().expandWildcardExpressions() == false
+            : "child request with action ["
+                + requestInfo.getAction()
+                + "] contains unexpanded wildcards "
+                + Arrays.stream(indices).filter(Regex::isSimpleMatchPattern).toList();
+
```

Comment:
This does not really represent the conditions under which a proper child-action contains unexpanded wildcards.
The condition reflects the fact that an action can be double-authorized.